src/Stratis.FederatedPeg.Features.FederationGateway/LeaderProvider.cs

```diff
@@ -23,24 +23,24 @@ public class LeaderProvider : ILeaderProvider
         /// </summary>
         private readonly IReadOnlyList<string> orderedFederationPublicKeys;
 
-        private PubKey currentLeader;
-
         public LeaderProvider(IFederationGatewaySettings federationGatewaySettings)
         {
             this.orderedFederationPublicKeys = federationGatewaySettings.FederationPublicKeys.
                 Select(k => k.ToString()).
                 OrderBy(j => j).
                 ToList().
                 AsReadOnly();
+
+            this.CurrentLeader = new PubKey(this.orderedFederationPublicKeys.First());
```

Comment:
Also here `this.CurrentLeader` is null at startup.  Now it is assigned to a fed member.